### ISSUE TITLE
Building of an Android Project fails when using Visual Studio 2022

### ISSUE THREAD

**Issue by LB-KacperSzczepanski** [o3de/o3de#11972]
**Describe the bug**
The building of the android project from command prompt ends with failure when using Visual Studio 2022. When building with Visual Studio 2022 a few different configurations were tried for making android debug build but the outcome was the same for all of them with an exception of asset number.
```
"> A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable

   > Zip64 forbidden but required in entry assets/gem/pythontests/streaming/benchmark/assets/10mb_4x3_0000003c.benchmark size=10486486, csize=10489274, cdloc=(offset=4,300,272,297, size=10,489,381), loc=(offset=4,300,272,404, size=10,489,274)"
```
The building was verified with 2 different source windows builds configured as debug and profile. Then the commands from the following guide were used (https://www.o3de.org/docs/user-guide/platforms/android/generating_android_project_windows/) with following changes:

- directories changed to indicate the engine and AutomatedTesting project built folders.
- Asset Processor Batch building command.
`"C:\o3de-build\android>cmake --build %O3DE_BUILD_ROOT%\windows --target AssetProcessorBatch --config debug -- -m /nologo"`
- command to build the Android Project.
`"cd %O3DE_BUILD_ROOT%\android gradlew assembleDebug"`
In all of those tests the outcome was the same. Please refer to the attached files for more information.

**Steps to reproduce**
1.   build the source engine using Visual Studio 2022 17.3.3.
2.   Follow the guide on  official site including the
prerequisites (https://www.o3de.org/docs/user-guide/platforms/android/generating_android_project_windows/).

**Expected behavior**
The Android Project is built and can be deployed to the android device.

**Actual behavior**
Building of the android project ends with an error for failing the execution of the task “app:packageProfile” or “app:packageDebug” depending on command used.

**Screenshot**
![image](https://user-images.githubusercontent.com/86953282/190176034-35023b77-f2fe-46ab-ba5b-1fccaf5a6e2f.png)

**Found in Branch**
Stabilization/2210 (441e2b3) - Stabilization/2210 (04e3edb)

 **Desktop/Device**
• Device: PC
• OS: Windows
• Version: 10 Pro
• CPU: Intel Core i7 11700F
• GPU: Nvidia RTX 3070 LHR
• Memory: 32GB

**Additional context**
Visual Studio 2022 17.3.3
Gradle 7.0.2
Android Studio Bumblebee 2021.1.1 Patch 3
SDK android-30
cmake 3.22.3
ninja 1.10.2
[info_log.txt](https://github.com/o3de/o3de/files/9567052/info_log.txt)
[scan_log.txt](https://github.com/o3de/o3de/files/9567053/scan_log.txt)

**Comment by LB-KacperSzczepanski** [comment]
Issue did not reproduce when project was build through the terminal in Android Studio on build Stabilization/2210 (37c7b44).

**Comment by LB-KacperSzczepanski** [comment]
Issue did not reproduce when using Visual Studio 2019 16.11.19  with build Stabilization/2210 (https://github.com/o3de/o3de/commit/37c7b44ceb096d502692bed0b9b38223a682514a).

**Comment by forhalle** [comment]
@lumberyard-employee-dm - Are you able to take a look at this?

**Comment by spham-amzn** [comment]
> Visual Studio doesn't have anything to do with the building of Android. Visual studio is only used to build the AssetProcessorBatch, which is used to process the assets. When it comes to building Android, that is using Gradle and ninja.
> 
> The issue here is that the APK is trying to produce a zip file with more than 4GiB worth of assets using regular Zip which only supports 32-bit or 4GiB filesize.
> 
> This warning
> 
> > Zip64 forbidden but required in entry assets/gem/pythontests/streaming/benchmark/assets/10mb_4x3_0000003c.benchmark size=10486486, csize=10489274, cdloc=(offset=4,300,272,297, size=10,489,381), loc=(offset=4,300,272,404, size=10,489,274)
> 
> Is indicating that the first file past the 2^32 limit(4,294,967,296) limit can't fit in a regular zip file, but Zip64 is needed.
> 
> An alternative is to add enable [zip64 support](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.bundling.Jar.html#org.gradle.api.tasks.bundling.Jar:zip64) in Gradle by adding the [zip64 option](https://discuss.gradle.org/t/better-documentation-for-zip64-option/13675) to build.gradle file.
> 
> @spham-amzn Looking at android_support.py, I see there are entries to use a `build.gradle.in` as a template, but I don't see that file anywhere within the code base(https://github.com/o3de/o3de/blob/development/cmake/Tools/Platform/Android/android_support.py#L1345). Is there a build.gradle.in file?

That file is here: https://github.com/o3de/o3de/blob/development/Code/Tools/Android/ProjectBuilder/build.gradle.in

**Comment by lumberyard-employee-dm** [comment]
I'll make a PR with adding the zip64 true change within that file.

**Comment by spham-amzn** [comment]
Although, a 4 GB APK is concerning

**Comment by LB-KacperSzczepanski** [comment]
Issue verified as Fixed in build Stabilization/2210 (b83f535).
